### DISCUSSION TITLE
Trial ends only after correct response but record all keypresses

### DISCUSSION THREAD

**Discussion by revatis2** [jspsych/jsPsych#2578] [Q&A]
Hi,
I have a design where I need participants to indicate switches in bi-stable perceptions. After they settle on one of the 2 perceptions, they should press a key to end the trial. Let's suppose they indicate percept 1 by pressing L, percept 2 by A, and after settling on one percept they press SPACE. Essentially, I want the trial to end on SPACE but before they press it, I want the data of switches (L-A) to be recorded too. 
How do I go about coding that?

`
var trial = {
    type: 'image-keyboard-response',
    stimulus: 'img/ambi.png',
    choices: ['l', 'a', 32],
    prompt: "Press SPACE when you retain one of the percepts",
};
`
In this code I want the trial to end only after pressing SPACE while recording other keypresses as well.

Thanks,
Revati

**Comment by jodeleeuw** [comment]
I'm going to convert this to a discussion.

**Comment by Qiuishere** [comment]
I think [loop_function ](https://www.jspsych.org/7.0/overview/timeline/#looping-timelines) can achieve your function. I have a similar structure here #2580 
In your case, you don't need to change any properties of the stimuli, but simply use the loop_node to record every response. Everytime a key is pressed, a new row will be created in the data file
